pkg/store/redis/store.go

```diff
@@ -34,20 +35,73 @@ func (rs *redisStore) SaveScanJob(scanJob job.ScanJob) error {
 	conn := rs.pool.Get()
 	defer rs.close(conn)
 
-	bytes, err := json.Marshal(scanJob)
+	scanJobJSON, err := rs.toJSON(scanJob)
 	if err != nil {
-		return xerrors.Errorf("marshalling scan job: %w", err)
+		return err
 	}
 
 	key := rs.getKeyForScanJob(scanJob.ID)
-	_, err = conn.Do("SET", key, string(bytes))
+
+	log.WithFields(log.Fields{
+		"scan_job_id":     scanJob.ID,
+		"scan_job_status": scanJob.Status.String(),
+		"redis_key":       key,
+	}).Debug("Saving scan job")
+
+	_, err = conn.Do("SET", key, scanJobJSON)
 	if err != nil {
 		return xerrors.Errorf("saving scan job: %w", err)
 	}
 
 	return nil
 }
 
+func (rs *redisStore) saveScanJobWithExpire(scanJob job.ScanJob, expire time.Duration) error {
+	conn := rs.pool.Get()
+	defer rs.close(conn)
+
+	scanJobJSON, err := rs.toJSON(scanJob)
+	if err != nil {
+		return err
+	}
+
+	key := rs.getKeyForScanJob(scanJob.ID)
+
+	log.WithFields(log.Fields{
+		"scan_job_id":     scanJob.ID,
+		"scan_job_status": scanJob.Status.String(),
+		"expire":          expire.String(),
+		"redis_key":       key,
+	}).Debug("Saving scan job with expire")
+
+	err = conn.Send("MULTI")
+	if err != nil {
+		return err
+	}
+	err = conn.Send("SET", key, scanJobJSON)
+	if err != nil {
+		return err
+	}
+	err = conn.Send("EXPIRE", key, int(expire.Seconds()))
+	if err != nil {
+		return err
+	}
+	_, err = conn.Do("EXEC")
+	if err != nil {
+		return xerrors.Errorf("saving scan job: %w", err)
+	}
+
+	return nil
+}
+
+func (rs *redisStore) toJSON(scanJob job.ScanJob) (string, error) {
+	bytes, err := json.Marshal(scanJob)
+	if err != nil {
+		return "", xerrors.Errorf("marshalling scan job: %w", err)
+	}
+	return string(bytes), nil
+}
```

Comment:
I'm not convinced this is worth a separate function, just to do json.Marshal and a cast to string - might as well just do this inline and save a few lines of code

Reply:
Yup. Also looking at error handling, I've effectively introduced an alias for json.Marshall named toJSON. It's gone already.